src/rules/spaceBeforeFunctionParenRule.ts

```diff
@@ -61,130 +61,64 @@ export class Rule extends Lint.Rules.AbstractRule {
+        const option = getOption(node, options);
+        if (option !== undefined) {
+            const openParen = Lint.childOfKind(node, ts.SyntaxKind.OpenParenToken)!;
+            const hasSpace = sourceFile.text[openParen.pos] !== "(";
+
+            if (hasSpace && option === "never") {
+                const pos = openParen.getStart() - 1;
+                ctx.addFailureAt(pos, 1, Rule.INVALID_WHITESPACE_ERROR, Lint.Replacement.deleteText(pos, 1));
+            } else if (!hasSpace && option === "always") {
+                const pos = openParen.getStart();
+                ctx.addFailureAt(pos, 1, Rule.MISSING_WHITESPACE_ERROR, Lint.Replacement.appendText(pos, " "));
             }
-        });
-    }
+        }
 
-    private getOption(optionName: OptionName): Option | undefined {
-        return this.cachedOptions[optionName];
-    }
+        ts.forEachChild(node, cb);
+    });
+}
 
-    private evaluateRuleAt(openParen?: ts.Node, option?: Option): void {
-        if (openParen === undefined || option === undefined) {
-            return;
-        }
+function getOption(node: ts.Node, options: Options): Option | undefined {
+    switch (node.kind) {
+        case ts.SyntaxKind.ArrowFunction:
+            return Lint.hasModifier(node.modifiers, ts.SyntaxKind.AsyncKeyword) ? options.asyncArrow : undefined;
 
-        const hasSpace = this.isSpaceAt(openParen.getStart() - 1);
+        case ts.SyntaxKind.Constructor:
+            return options.constructor;
 
-        if (hasSpace && option === "never") {
-            const pos = openParen.getStart() - 1;
-            this.addFailureAt(pos, 1, Rule.INVALID_WHITESPACE_ERROR, this.deleteText(pos, 1));
-        } else if (!hasSpace && option === "always") {
-            const pos = openParen.getStart();
-            this.addFailureAt(pos, 1, Rule.MISSING_WHITESPACE_ERROR, this.appendText(pos, " "));
-        }
-    }
+        case ts.SyntaxKind.FunctionDeclaration:
+            return options.named;
 
-    private isSpaceAt(textPos: number): boolean {
-        this.scanner.setTextPos(textPos);
-        const prevTokenKind = this.scanner.scan();
-        return prevTokenKind === ts.SyntaxKind.WhitespaceTrivia;
-    }
+        case ts.SyntaxKind.FunctionExpression:
+            return (node as ts.FunctionExpression).name !== undefined ? options.named : options.anonymous;
 
-    private visitFunction(node: ts.Node): void {
-        const identifier = Lint.childOfKind(node, ts.SyntaxKind.Identifier);
-        const hasIdentifier = identifier !== undefined && (identifier.getEnd() !== identifier.getStart());
-        const optionName = hasIdentifier ? "named" : "anonymous";
-        const option = this.getOption(optionName);
-        const openParen = Lint.childOfKind(node, ts.SyntaxKind.OpenParenToken);
-        this.evaluateRuleAt(openParen, option);
-    }
+        case ts.SyntaxKind.MethodDeclaration:
+        case ts.SyntaxKind.MethodSignature:
+            return options.method;
```

Comment:
could you add `SetAccessor` and `GetAccessor` while you are already editing this rule?